private/model/api/shape.go

```diff
@@ -201,6 +210,9 @@ func (s *Shape) GoType() string {
 // GoType returns a shape ref's Go type.
 func (ref *ShapeRef) GoType() string {
 	if ref.Shape == nil {
+		if ref.JSONValue {
+			return "interface{}"
```

Comment:
Yea, apparently during one of the traversals it walks down to the map value. However, if we don't need to walk the map, it might fix this.